### PR TITLE
Phone number in rat profile not saving if validation does not pass (EXPOSUREAPP-12964)

### PR DESCRIPTION
**Pull request by SamuraiKek**
The current implementation of the validation for the phone number and email fields was causing the data to be deleted if the fields were edited and the validation didn't pass. Since these fields are not required for the profile creation I've removed the validation altogether. The appropriate keyboards show up when typing so the formatting of the text should still work.

### CODE REVIEW

**Review by mtwalli — DISMISSED** [review]

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/profile/ui/create/ProfileCreateFragment.kt [hunk 1]
```diff
@@ -82,39 +81,13 @@ class ProfileCreateFragment : Fragment(R.layout.profile_create_fragment), AutoIn
...
-                    phoneInputLayout.error = null
-                }
+                viewModel.phoneChanged(it.toString())
             }
             phoneInputEdit.addTextChangedListener(PhoneNumberFormattingTextWatcher())
 
             // E-mail
             emailInputEdit.addEmojiFilter().doAfterTextChanged {
-                // Propagate email to view model if it matches the pattern
-                if (Patterns.EMAIL_ADDRESS.matcher(it.toString()).matches()) {
```
Comment (schauersbergern) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/profile/ui/create/ProfileCreateFragment.kt:105]: Shouldn't we keep the e-mail validation at least, to have a valid e-mail? When I scan the QR from the profile it also displays the E-Mail field. Maybe hospitals/officials would use this

**Comment by MikeMcC399** [comment]
I successfully validated against the steps to reproduce in #5130:

## Invalid phone number not saved

1. Click on Manage Your Tests
2. Scroll down and click on Rapid Test Profiles
3. Click "+ RAPID TEST PROFILE"
4. Enter "A" in "First Name" then "9" in "Phone Number"
5. Click "Save"
6. "9" is stored and shown

## Trailing space makes phone number invalid

1. Click on Manage Your Tests
2. Scroll down and click on Rapid Test Profiles
3. Click "+ RAPID TEST PROFILE"
4. Enter a valid telephone number, such as "08007540001" in "Phone Number" then add a space after the digits
5. Click "Save" button
6. The phone number "08007540001" is shown


An enhancement to this UI would be to strip off any trailing blanks. This isn't strictly necessary though if the field verification has been removed. This PR does resolve #5130 in its current form.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/profile/ui/create/ProfileCreateFragment.kt (hunk 1 above)
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/profile/ui/create/ProfileCreateFragment.kt:105]: Alright, I will keep the validation for the email.

Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/profile/ui/create/ProfileCreateFragment.kt:105]: Updated the validation. In the end I also added it for the phone number as it wasn't much trouble.

**Comment by SamuraiKek** [comment]
@MikeMcC399, I've updated the fix to actually not let you save the phone number & email until it's formatted correctly (this also takes care of any trailing spaces).

**Comment by MikeMcC399** [comment]
@SamuraiKek 

> I've updated the fix to actually not let you save the phone number & email until it's formatted correctly (this also takes care of any trailing spaces).

If the user enters a phone number, actively entering `<space>` characters, for example:
`+49<space>172<space>12345678`
there is still an unexpected trailing space left in the field causing "Invalid phone number". Since the trailing space is not immediately visible unless the user moves the cursor to the end of the field, it is difficult to spot the problem.

It is however a definite improvement from the original condition!

**Review by schauersbergern — APPROVED** [review]

**Review by mtwalli — APPROVED** [review]

**Pull request merged**